crates/evm/src/lib.rs

```diff
@@ -23,6 +24,18 @@ pub mod provider;
 /// test helpers for mocking executor
 pub mod test_utils;
 
+/// Trait for configuring the EVM in a generic way
+pub trait ConfigureEvmGeneric: ConfigureEvm + ConfigureEvmEnv {
+    /// Returns a new EVM with the given database configured with the given environment settings.
+    fn evm_with_env_ext<'a, DB>(
+        &'a self,
+        db: DB,
+        env: EnvWithHandlerCfg,
+    ) -> Box<dyn EvmTransact<DB = DB> + 'a>
```

Comment:
also do we have to return a trait object? Can't you add a generic `EVM: EvmTransact<DB = DB>`?

Reply:
In this situation there are inference problems if we choose the approach you mention and all the compiler's suggestions converge towards the use of a trait object I have the impression...